### PR TITLE
Add sample of identify and merge cross-page tables

### PR DESCRIPTION
**Pull request by TakaValley**
Identify and merge cross page markdown tables

### CODE REVIEW

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb [hunk 1]
```diff
@@ -4,17 +4,28 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Identify cross-page tables based on rules\n",
+    "# Identify cross-page tables and merge them into one table based on rules\n",
     "\n",
```
Comment (luzhang06) [inline SampleCode/Python/sample_identify_cross_page_tables.ipynb:8]: merge them into one table in the markdown output

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb [hunk 2]
```diff
@@ -4,17 +4,28 @@
...
-    "This sample demonstrates how to use the output of Layout model and some business rules to identify cross-page tables. Once idenfied, it can be further processed to merge these tables and keep the semantics of a table.\n",
+    "This sample demonstrates how to use the output of Layout model and some business rules to identify markdown cross-page tables. Once idenfied, it will be processed to merge these markdown tables and keep the semantics of a table.\n",
     "\n",
```
Comment (luzhang06) [inline SampleCode/Python/sample_identify_cross_page_tables.ipynb:10]: it's not identify markdown cross-page tables, it's identify cross-page tables and merge these tables in the markdown output.

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb [hunk 3]
```diff
@@ -4,17 +4,28 @@
...
-    "Depending on your document format, there can be different rules applied to idenfity a cross-page table. This sample shows how to use the following rules to identify cross-page tables:\n",
+    "Depending on your document format, there can be different rules applied to idenfity a cross-page markdown table. This sample shows how to use the following rules to identify cross-page markdown tables:\n",
     "\n",
```
Comment (luzhang06) [inline SampleCode/Python/sample_identify_cross_page_tables.ipynb:12]: same here. it's not identify cross-page markdown table.

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb [hunk 4]
```diff
@@ -4,17 +4,28 @@
...
+    "    - If there're 2 or more tables appear in consecutive pages\n",
+    "    - And the table's right side is quite close to the right edge of current page\n",
+    "    - And the next table's left side is quite close to the left edge of next page\n",
+    "    - And the tables have the same number of row count\n",
+    "    - These tables could be considered to one horizontal table.\n",
+    "\n",
+    "You can customize the rules based on your scenario.\n",
+    "\n",
+    "**Tip** : In future versions, we will use html to represent tables in markdown soon since markdown doesn't support merged cell. If you run into merged cell issues, you can consider using html as the solution. A reference code can be found here: *https://github.com/anirudhabodhankar/FormRecognizer-/tree/main/Form-Recognizer-Toolkit* ."
    ]
```
Comment (luzhang06) [inline SampleCode/Python/sample_identify_cross_page_tables.ipynb:29]: Please remove the tip part for now. We will make announcement and update this sample once we release the html format.

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb [hunk 5]
```diff
@@ -128,28 +145,51 @@
...
+    "            pre_max_offset = max_offset\n",
+    "        else:\n",
+    "            print(f\"Table {table_idx} is empty\")\n",
+    "            table_concision_list.append(\n",
+    "                {\"idx\": {table_idx}, \"min_offset\": -1, \"max_offset\": -1, \"page\": -1}\n",
+    "            )\n",
     "\n",
-    "    return merge_tables_candidates"
+    "    return merge_tables_candidates, table_concision_list"
    ]
```
Comment (luzhang06) [inline SampleCode/Python/sample_identify_cross_page_tables.ipynb:193]: what does table_concision_list mean? can you please help explain a bit?

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb [hunk 6]
```diff
@@ -230,7 +491,15 @@
...
+    "if __name__ == \"__main__\":\n",
+    "    # Check if the input string is provided as a command-line argument\n",
+    "    if len(sys.argv) > 1:\n",
+    "        input_file_path = sys.argv[1]\n",
+    "    else:\n",
+    "        print(\"Usage: python sample_identify_cross_page_tables.py [input_file_path]\")\n",
+    "        sys.exit(1)\n",
+    "\n",
+    "    identify_and_merge_cross_page_tables(input_file_path)"
    ]
```
Comment (luzhang06) [inline SampleCode/Python/sample_identify_cross_page_tables.ipynb:503]: how do you run this in the ipynb? My approach is just call the function directly "identify_and_merge_cross_page_tables(input_file_path)". maybe I'm not aware of this approach in jupyter notebook.

File: SampleCode/Python/sample_identify_cross_page_tables.py [hunk 7]
```diff
@@ -161,32 +326,100 @@ def identify_cross_page_tables(input_file_path):
...
+
+    
+    optimized_content= ""
+    if merged_table_list:
+        print(f"Merged {len(merged_table_list)} markdown tables totally.")
+        print("=========================================================")
+        start_idx = 0
+        for merged_idx, merged_table in enumerate(merged_table_list):
+            print(f"Hight the merged table {merged_idx}")
+            print("-----------------------------------------------------")
```
Comment (luzhang06) [inline SampleCode/Python/sample_identify_cross_page_tables.py:404]: Highlight?

Comment (luzhang06) [reply SampleCode/Python/sample_identify_cross_page_tables.py:404]: the merged idx is not very straight forward - can we reference the original table index so we know which ones are merged?

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb (hunk 1 above)
Comment (TakaValley) [reply SampleCode/Python/sample_identify_cross_page_tables.ipynb:8]: fixed

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb (hunk 4 above)
Comment (TakaValley) [reply SampleCode/Python/sample_identify_cross_page_tables.ipynb:29]: fixed

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb (hunk 2 above)
Comment (TakaValley) [reply SampleCode/Python/sample_identify_cross_page_tables.ipynb:10]: Fixed.

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb (hunk 5 above)
Comment (TakaValley) [reply SampleCode/Python/sample_identify_cross_page_tables.ipynb:192]: Fixed.

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb (hunk 3 above)
Comment (TakaValley) [reply SampleCode/Python/sample_identify_cross_page_tables.ipynb:12]: Fixed.

File: SampleCode/Python/sample_identify_cross_page_tables.ipynb (hunk 6 above)
Comment (TakaValley) [reply SampleCode/Python/sample_identify_cross_page_tables.ipynb:495]: Fixed. Thx~
JupyterLabScreen

File: SampleCode/Python/sample_identify_cross_page_tables.py (hunk 7 above)
Comment (TakaValley) [reply SampleCode/Python/sample_identify_cross_page_tables.py:407]: “Highlight” is inappropriate.  Refactored it follow the suggestion. Thx

**Review by toothache — APPROVED** [review]
LGTM

**Pull request merged**